clang/lib/CodeGen/SanitizerMetadata.cpp

```diff
@@ -105,11 +106,32 @@ void SanitizerMetadata::reportGlobal(llvm::GlobalVariable *GV,
                                            GV, Loc, Ty, "init");
 
   GV->setSanitizerMetadata(Meta);
+
+  if (Ty.isNull() || !CGM.getLangOpts().Sanitize.has(SanitizerKind::Type) ||
```

Comment:
We also need to exclude globals defined externally as well I think: https://github.com/llvm/llvm-project/pull/120565